docs/src/tutorials/methods/cellarea.jl

```diff
@@ -0,0 +1,114 @@
+#
+all_countries = naturalearth("admin_0_countries", 10)
+chile = all_countries.geometry[findfirst(==("Chile"), all_countries.NAME)]
+# Let's plot the precipitation on the world map, and highlight Chile:
+f, a, p = heatmap(precip; colorrange = Makie.zscale(replace_missing(precip, NaN)), axis = (; aspect = DataAspect()))
+p2 = poly!(a, chile; color = (:red, 0.3), strokecolor = :red, strokewidth = 0.5)
+f
+# You can see Chile highlighted in red, in the bottom left quadrant.
+#
+# First, let's make sure that we only have the data that we care about, and crop and mask the raster so it only has values in Chile.
+# We can crop by the geometry, which really just generates a view into the raster that is bounded by the geometry's bounding box.
+cropped_precip = crop(precip; to = chile)
+# Now, we mask the data such that any data outside the geometry is set to `missing`.
+masked_precip = mask(cropped_precip; with = chile)
+heatmap(masked_precip)
+# This is a lot of missing data, but that's mainly because the Chile geometry we have encompasses the Easter Islands as well, in the middle of the Pacific.
+
+# Now, let's compute the average precipitation per square meter across Chile.
+# First, we need to get the area of each cell in square meters.  We'll use the spherical method, since we're working with a geographic coordinate system.  This is the default.
+areas = cellarea(masked_precip)
+masked_areas = mask(areas; with = chile)
+heatmap(masked_areas; axis = (; title = "Cell area in square meters"))
+# Now we can compute the average precipitation per square meter.  First, we compute total precipitation per grid cell:
+precip_per_area = masked_precip .* masked_areas
+# We can sum this to get the total precipitation per square meter across Chile:
+total_precip = sum(skipmissing(precip_per_area))
+# We can also sum the areas to get the total area of Chile (in this raster, at least).
+total_area = sum(skipmissing(masked_areas))
+# And we can convert that to an average by dividing by the total area:
+avg_precip = total_precip / total_area
+# According to the internet, Chile gets about 100mm of rain per square meter in June, so our statistic seems pretty close.
+#
+# Let's see what happens if we don't account for cell areas:
+bad_total_precip = sum(skipmissing(masked_precip))
+bad_avg_precip = bad_total_precip / length(collect(skipmissing(masked_precip)))
+# This is misestimated!  This is why it's important to account for cell areas when computing averages.
+
+#=
+!!! note
+    If you made it this far, congratulations!  
+
+    It's interesting to note that we've replicated the workflow of `zonal` here.  
+    `zonal` is a more general function that can be used to compute any function over geometries, 
+    and it has multithreading built in.  
+    
+    But fundamentally, this is all that `zonal` is doing under the hood - 
+    masking and cropping the raster to the geometry, and then computing the statistic.
+=#
```

Comment:
from experience (a painful one), I can tell you that for long explanations using literate and `.jl` files is not ideal. I did that for a while in the yax docs, and eventually I went back to just markdown. For small scripts that just run is fine, like in beautifulmakie, but once you start doing bigger things I would say that markdown should be the way to go.

Reply:
Hmm, why not?  It's worked pretty well for me so far, but those were all in situations where my markdown content doesn't exceed ~150 lines.  Curious what your issues were...

Reply:
I do at some point want to integrate a Quarto like system here, since that seems to be the best at this stuff.

Reply:
I agree with @lazarusA, Literate.jl adds a layer of complexity that's just not worth it in some cases. 

I would prefer not to have it here